sansio_lsp_client/client.py

```diff
@@ -171,31 +360,63 @@ def parse_request(event_cls: t.Type[Event]) -> Event:
                     " or ServerNotification"
                 )
 
-        if request.method == "window/showMessage":
+        if request.method == "workspace/workspaceFolders":
+            return parse_request(WorkspaceFolders)
+
+        elif request.method == "workspace/configuration":
+            return parse_request(ConfigurationRequest)
+
+        elif request.method == "window/showMessage":
             return parse_request(ShowMessage)
         elif request.method == "window/showMessageRequest":
             return parse_request(ShowMessageRequest)
         elif request.method == "window/logMessage":
             return parse_request(LogMessage)
+
         elif request.method == "textDocument/publishDiagnostics":
             return parse_request(PublishDiagnostics)
-        else:
-            raise NotImplementedError(request)
 
-    def recv(self, data: bytes) -> t.List[Event]:
+        elif request.method == "window/workDoneProgress/create":
+            ev = parse_request(WorkDoneProgressCreate)
+            self._progress_tokens_map[request.params['token']] = WorkDoneProgress
+            return parse_request(WorkDoneProgressCreate)
+
+        elif request.method == "$/progress":
+            progress_type = self._progress_tokens_map.get(request.params['token'])
+
+            if progress_type == WorkDoneProgress:
+                kind = request.params.get('value', {}).get('kind')
```

Comment:
Does this work correctly if `kind` is `None`?

Reply:
`kind` can only be 'begin', 'report' and 'end' ... or is there a chance of it being expanded? 

For now there will be a `NotImplementedError(request)` in that case. 

I'm open to advice on this matter :)

Reply:
Maybe you meant `request.params['value']['kind']`? That errors when the kind is missing, and doesn't produce unexpected `None`s.